core/src/ast/methods.rs

```diff
@@ -383,8 +383,8 @@ mod tests {
             );
 
             let borrowed_params = method.borrowed_params();
-            let actual: Vec<&str> = borrowed_params.names("self").collect();
-            let expected: &[&str] = &[$(stringify!($param)),*];
+            let actual: Vec<&Ident> = borrowed_params.names(&Ident::THIS).collect();
```

Comment:
question: this moves us from self to this in diplomat_core (which shouldn't need to worry about backend specific things like `this`). intentional?

Reply:
I fixed this in "Fixes tests" cause this actually broke it lol. My fix wasn't that clean though so lmk if you want to me to reconsider